src/org/pentaho/reporting/platform/plugin/async/IJobIdGenerator.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License, version 2 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/gpl-2.0.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ *
+ * Copyright 2006 - 2016 Pentaho Corporation.  All rights reserved.
+ */
+
+package org.pentaho.reporting.platform.plugin.async;
+
+import org.pentaho.platform.api.engine.IPentahoSession;
+
+public interface IJobIdGenerator {
+
+  String generateId( IPentahoSession session );
```

Comment:
Can this be strongly typed? We expect an UUID in the string, so let it be a UUID in the interface too. On the caller-site, you can easily parse the UUID via "UUID.fromString()" - which also validates the input. If the UUID given there is non-null, but invalid (ie illegal-arg exception is thrown), abort with an error. Aborting on invalid input is OK and safe and sane.